### PR TITLE
SDK AuthZ API

### PR DESCRIPTION
**Pull request by Buckram123**
~~MERGE #147 FIRST~~
### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: framework/packages/abstract-sdk/src/apis/authz.rs [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+}
+
+impl GenericAuthorization {
+    /// Create new generic authorization
+    pub fn new(msg: String) -> Self {
+        Self { msg }
+    }
+}
+
+trait AuthZAuthrization {
```
Comment (CyberHoward) [inline framework/packages/abstract-sdk/src/apis/authz.rs:135]: What's the goal/point of this trait?

Comment (CyberHoward) [reply framework/packages/abstract-sdk/src/apis/authz.rs:135]: Would be nice to have something that also accepts raw proto messages (`Vec<u8>`)

File: framework/packages/abstract-sdk/src/apis/authz.rs [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    /// Grants generic authorization to a **grantee**.
+    ///
+    /// # Arguments
+    ///
+    /// * `grantee` - The address of the grantee.
+    /// * `msg` - Allowed message type url.
+    pub fn grant_generic(
+        &self,
+        grantee: &Addr,
+        msg: String,
```
Comment (CyberHoward) [inline framework/packages/abstract-sdk/src/apis/authz.rs:95]:
```suggestion
        msg_type_url: String,
```

File: framework/packages/abstract-sdk/src/apis/authz.rs (hunk 1 above)
Comment (CyberHoward) [reply framework/packages/abstract-sdk/src/apis/authz.rs:135]: If there are other grant types (besides GenericAuthorization) then we should also include those in here. And imo that should be in a nested mod

```rust
mod authorizations {
    // trait

    // different impls
}
```

Comment (Buckram123) [reply framework/packages/abstract-sdk/src/apis/authz.rs]: > What's the goal/point of this trait?

It helps structuring different types of grants

> If there are other grant types (besides GenericAuthorization) then we should also include those in here. And imo that should be in a nested mod
I will add other types of grants. Yeah, nested mod is fine

**Review by CyberHoward — APPROVED** [review]
LGTM!

**Pull request merged**